### PR TITLE
Fixes/app store keys in db

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

Implements App model to save App Store registry and secrets so we can keep our `.env` variables under the [4kb size limit imposed by AWS](https://vercel.com/docs/concepts/limits/overview#environment-variables).

### TODO:
- ~~Save as much metadata as needed in DB~~
- ~~Create a migration to relate `Credentials` to their corresponding `App`~~

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How should this be tested?

- [ ] App Store apps should work as before

### CODE REVIEW

File: apps/web/lib/attendeeToPersonConversionType.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
+import { Attendee } from "@prisma/client";
 import { TFunction } from "next-i18next";
 
-import { Attendee } from "@calcom/prisma/client";
```
Comment (zomars) [inline apps/web/lib/attendeeToPersonConversionType.ts:3]: These were throwing type errors

File: apps/web/next.config.js [hunk 2]
```diff
@@ -104,6 +104,28 @@ const nextConfig = {
...
+          destination: "/apps/huddle01",
+          permanent: true,
+        },
+        {
+          source: "/apps/jitsi_video",
+          destination: "/apps/jitsi",
+          permanent: true,
+        }
+      );
+    }
```
Comment (zomars) [inline apps/web/next.config.js:126]: To prevent already 404s for already published links

**Review by zomars — COMMENTED** [review]
Ready for review @alannnc @agustif

**Review by agustif — APPROVED** [review]
Awesome work as always @zomars !!

I only tested the giphy app w extra credentials and worked like a charm
I did fail first time by adding it to web/.env instead of root's .env so migration wouldn't pick it.

Awesome work and clearing the path for the hosted apps in the future!

**Review by alannnc — APPROVED** [review]
Code looks good, haven't tested it properly yet.

File: apps/web/pages/d/[link]/book.tsx [hunk 3]
```diff
@@ -173,7 +153,7 @@ export async function getServerSideProps(context: GetServerSidePropsContext) {
       locationLabels: getLocationLabels(t),
       profile,
       eventType: eventTypeObject,
-      booking,
+      booking: null,
```
[the remark this answers is not in the document]
Comment (zomars) [reply apps/web/pages/d/[link]/book.tsx:156]: Yes, it's also resolved on @alishaz-polymath 's PR #2644

**Pull request merged**